src/core/packet.lua

```diff
@@ -20,10 +18,40 @@ local header_size = 8
 local max_payload = tonumber(C.PACKET_PAYLOAD_SIZE)
 
 -- Freelist containing empty packets ready for use.
+
+ffi.cdef[[
+struct freelist {
+    int nfree, max;
+    struct packet *list[?];
+};
+]]
```

Comment:
Sounds good to me, only thing is I'm not sure that __align__((64)) works with ffi.new.  It certainly wouldn't work if it were just malloc.  Incidentally, why do you consider alignment to be important if we are always working on the back of the list, a dynamic location which will likely be unaligned?  I also note that it's not the number of links in the network but the number of places the packet ingresses and egresses from Snabb.  For me I leave it up to @xrme to determine whether he wants to incorporate a change like this, I won't block the commit because of it, though I'm happy to accept follow-ups :)